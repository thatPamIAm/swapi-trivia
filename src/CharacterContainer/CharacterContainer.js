import React, { Component } from 'react';
import Character from '../Character/Character';

import './CharacterContainer.css'
class CharacterContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      characters: []
    }
  }

  componentDidMount() {
    this.fetchCharacters(this.props.characters)
  }

  async fetchCharacters(list) {
  let characters = await Promise.all(
    list.map(async url => {
      let character = await fetch(url);
      return character.json()
    })
  )

  this.setState({
    characters: characters
  })
 }

  render() {
    const characters = this.state.characters.map((character, index) => {
      return <Character character={character} key={index} />
    });

    return (
      <section className="container">
        {characters}
      </section>
    )  
  }
}

export default CharacterContainer;