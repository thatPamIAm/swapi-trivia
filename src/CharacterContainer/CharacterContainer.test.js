import React from 'react';
import { shallow } from 'enzyme';

import CharacterContainer from './CharacterContainer';

describe('CharacterContainer', () => {
  let wrapper;
  const mockUrls = ['https://swapi.co/api/people/2/',
                    'https://swapi.co/api/people/3/',
                    'https://swapi.co/api/people/10/'
                  ];

  beforeEach(() => {
    wrapper = shallow(<CharacterContainer urls={mockUrls} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should hae a default state', () => {
    expect(wrapper.state().characters).toEqual([]);
  }); 
});