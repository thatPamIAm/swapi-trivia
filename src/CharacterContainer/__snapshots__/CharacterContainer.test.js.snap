// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer should match the snapshot 1`] = `ShallowWrapper {}`;
